shared/src/business/useCases/users/generateChangeOfAddress.test.js

```diff
@@ -720,4 +722,115 @@ describe('generateChangeOfAddress', () => {
+      },
+      user: {
+        barNumber: 'PT5432',
+        contact: {
+          address1: '234 Main St!',
+          address2: 'Apartment 4',
+          address3: 'Under the stairs',
+          city: 'Chicago',
+          countryType: COUNTRY_TYPES.DOMESTIC,
+          phone: '+1 (555) 555-5555',
+          postalCode: '61234',
+          state: 'IL',
+        },
+        email: 'privatePractitioner1',
+        name: 'Test Private Practitioner',
+        representingPrimary: true,
+        role: ROLES.privatePractitioner,
+        section: 'privatePractitioner',
+        serviceIndicator: SERVICE_INDICATOR_TYPES.SI_ELECTRONIC,
+        userId: 'ad07b846-8933-4778-9fe2-b5d8ac8ad728',
+      },
+    });
+
+    expect(
+      applicationContext.getDocumentGenerators().changeOfAddress,
+    ).not.toHaveBeenCalled();
+    expect(
+      applicationContext.getPersistenceGateway().saveWorkItemForNonPaper,
+    ).not.toHaveBeenCalled();
+    expect(cases).toMatchObject([
+      expect.objectContaining({ docketNumber: MOCK_CASE.docketNumber }),
+    ]);
+    // We still want to update our case!
+    expect(
+      applicationContext.getPersistenceGateway().updateCase,
+    ).toHaveBeenCalled();
+  });
+
+  it('should create a docket entry, work item, and serve it if the case is closed less than six months ago, and it should still update the case', async () => {
+    mockCaseWithPrivatePractitioner.status = CASE_STATUS_TYPES.closed;
+    mockCaseWithPrivatePractitioner.closedDate = calculateISODate({
+      howMuch: -1,
+      units: 'months',
+    });
+
+    const cases = await generateChangeOfAddress({
+      applicationContext,
+      contactInfo: {
+        address1: '234 Main St',
+        address2: 'Apartment 4',
+        address3: 'Under the stairs',
+        city: 'Chicago',
+        countryType: COUNTRY_TYPES.DOMESTIC,
+        phone: '+1 (555) 555-5555',
+        postalCode: '61234',
+        state: 'IL',
+      },
+      user: {
+        barNumber: 'PT5432',
+        contact: {
+          address1: '234 Main St!',
+          address2: 'Apartment 4',
+          address3: 'Under the stairs',
+          city: 'Chicago',
+          countryType: COUNTRY_TYPES.DOMESTIC,
+          phone: '+1 (555) 555-5555',
+          postalCode: '61234',
+          state: 'IL',
+        },
+        email: 'privatePractitioner1',
+        name: 'Test Private Practitioner',
+        representingPrimary: true,
+        role: ROLES.privatePractitioner,
+        section: 'privatePractitioner',
+        serviceIndicator: SERVICE_INDICATOR_TYPES.SI_ELECTRONIC,
+        userId: 'ad07b846-8933-4778-9fe2-b5d8ac8ad728',
+      },
+    });
+
+    const docketEntryForNoticeOfChangeOfAddress = cases[0].docketEntries.find(
+      entry => entry.documentTitle.includes('Notice of Change'),
+    );
+
+    expect(
+      applicationContext.getDocumentGenerators().changeOfAddress,
+    ).toHaveBeenCalled();
+    expect(docketEntryForNoticeOfChangeOfAddress).toBeDefined();
```

Comment:
should this check if the `saveWorkItemForNonPaper` function was called also?

Reply:
Again the default `MOCK_CASE` has everyone on the case be `SI_ELECTRONIC`. For good measure, I have made the case in this test have the secondary contact who requires paper service and reintroduced this expect.

Now you have me wondering about that name of that function. It's literally doing the opposite... saving a work item for a case that requires paper service. I've logged a refactoring ticket to discuss further.